samza-core/src/main/java/org/apache/samza/clustermanager/AbstractContainerAllocator.java

```diff
@@ -33,16 +33,13 @@
 
 /**
  * {@link AbstractContainerAllocator} makes requests for physical resources to the resource manager and also runs
- * a container process on an allocated physical resource. Sub-classes should override the assignResourceRequests()
+ * a processor on an allocated physical resource. Sub-classes should override the assignResourceRequests()
  * method to assign resource requests according to some strategy.
  *
  * See {@link ContainerAllocator} and {@link HostAwareContainerAllocator} for two such strategies
  *
- * This class is not thread-safe.
+ * This class is not thread-safe. This class is used in the refactored code path as called by run-jc.sh
```

Comment:
Maybe update this, until we delete the AbstractContainerAllocator.
// This class is instantiated by the ContainerProcessManager (which inturn is created by the JC from run-jc.sh), when host-affinity is off. Otherwise, the HostAwareContainerAllocator is instantiated.

Reply:
Moved this comment to ContainerAllocator.